src/components/Select/SelectT.tsx

```diff
@@ -81,6 +81,33 @@ export default function SelectT<T>(props: SelectTProps<T>): JSX.Element {
   const inputRef = useRef<HTMLDivElement>(null);
   const dropdownRef = useRef<HTMLUListElement>(null);
 
+  const hasOptions = useMemo<boolean>(() => options !== undefined &&  options.length > 0, [options]);
+
+  const repositionMenu = useCallback(() => {
```

Comment:
These were moved up in order and converted to a useCallback. No other code changes here.